commit 169d7e5a922ef8e63b51ee2bdf4fd4ad60ed2689
Author: dmiller <dmiller@e0a8ed71-7df4-0310-8962-fdc924857419>
Date:   Thu Nov 19 17:34:24 2020 +0000

    Restore call to post_connect in non-ssl case. Fixes #2149

diff --git a/ncat/ncat_connect.c b/ncat/ncat_connect.c
index 606eac412..f59dd4372 100644
--- a/ncat/ncat_connect.c
+++ b/ncat/ncat_connect.c
@@ -1064,12 +1064,17 @@ int ncat_connect(void)
             bye("Failed to set hostname on iod.");
         if (o.ssl)
         {
+            /* connect_handler creates stdin_nsi and calls post_connect */
             nsock_reconnect_ssl(mypool, cs.sock_nsi, connect_handler, o.conntimeout, NULL, NULL);
         }
+        else
+        {
+            /* Create IOD for nsp->stdin */
+            if ((cs.stdin_nsi = nsock_iod_new2(mypool, 0, NULL)) == NULL)
+                bye("Failed to create stdin nsiod.");
 
-        /* Create IOD for nsp->stdin */
-        if ((cs.stdin_nsi = nsock_iod_new2(mypool, 0, NULL)) == NULL)
-            bye("Failed to create stdin nsiod.");
+            post_connect(mypool, cs.sock_nsi);
+        }
     }
 
     /* connect */
